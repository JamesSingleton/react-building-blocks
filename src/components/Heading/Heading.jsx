import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const APPEARANCES = {
  H1: 'H1',
  H2: 'H2',
  H3: 'H3',
  H4: 'H4',
  H5: 'H5',
  H6: 'H6',
};

// eslint-disable-next-line no-unused-vars
const StyledH1 = styled.h1`
  font-size: 2rem;
  line-height: 3rem;
  color: #162d3d;
`;

const StyledH2 = styled.h2`
  font-size: 1.75rem;
  line-height: 2rem;
  color: #162d3d;
`;

const StyledH3 = styled.h3`
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #162d3d;
`;

const StyledH4 = styled.h4`
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #162d3d;
`;

const StyledH5 = styled.h5`
  font-size: 0.75rem;
  line-height: 1.5rem;
  color: #32536a;
  text-transform: uppercase;
  letter-spacing: 1px;
`;

const StyledH6 = styled.h6`
  font-size: 0.625rem;
  line-height: 1.125rem;
  color: #162d3d;
  text-transform: uppercase;
  letter-spacing: 1px;
`;

const Heading = ({ appearance, children, ...rest }) => {
  const { dataHook, ...headingProps } = rest;
  if (appearance.toLowerCase() === 'h2') {
    return <StyledH2 {...headingProps}>{children}</StyledH2>;
  }
  if (appearance.toLowerCase() === 'h3') {
    return <StyledH3 {...headingProps}>{children}</StyledH3>;
  }
  if (appearance.toLowerCase() === 'h4') {
    return <StyledH4 {...headingProps}>{children}</StyledH4>;
  }
  if (appearance.toLowerCase() === 'h5') {
    return <StyledH5 {...headingProps}>{children}</StyledH5>;
  }
  if (appearance.toLowerCase() === 'h6') {
    return <StyledH6 {...headingProps}>{children}</StyledH6>;
  }
  return <StyledH1 {...headingProps}>{children}</StyledH1>;
};

Heading.propTypes = {
  children: PropTypes.oneOf([
    PropTypes.string,
    PropTypes.node,
    PropTypes.element,
  ]).isRequired,
  appearance: PropTypes.oneOf(Object.keys(APPEARANCES)),
};

Heading.defaultProps = {
  appearance: APPEARANCES.H1,
};

export default Heading;
