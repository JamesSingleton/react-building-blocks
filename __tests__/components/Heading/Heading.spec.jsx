import React from 'react';
import { mount } from 'enzyme';
import Heading from '../../../src/components/Heading';

describe('Heading', () => {
  it('should render as an H1', () => {
    const props = {
      appearance: 'H1',
    };
    const renderedHeading = mount(<Heading {...props}>Hello World</Heading>);
    expect(renderedHeading).toMatchSnapshot();
  });
  it('should render as an H2', () => {
    const props = {
      appearance: 'H2',
    };
    const renderedHeading = mount(<Heading {...props}>Hello World</Heading>);
    expect(renderedHeading).toMatchSnapshot();
  });
  it('should render as an H3', () => {
    const props = {
      appearance: 'H3',
    };
    const renderedHeading = mount(<Heading {...props}>Hello World</Heading>);
    expect(renderedHeading).toMatchSnapshot();
  });
  it('should render as an H4', () => {
    const props = {
      appearance: 'H4',
    };
    const renderedHeading = mount(<Heading {...props}>Hello World</Heading>);
    expect(renderedHeading).toMatchSnapshot();
  });
  it('should render as an H5', () => {
    const props = {
      appearance: 'H5',
    };
    const renderedHeading = mount(<Heading {...props}>Hello World</Heading>);
    expect(renderedHeading).toMatchSnapshot();
  });
  it('should render as an H6', () => {
    const props = {
      appearance: 'H6',
    };
    const renderedHeading = mount(<Heading {...props}>Hello World</Heading>);
    expect(renderedHeading).toMatchSnapshot();
  });
});
