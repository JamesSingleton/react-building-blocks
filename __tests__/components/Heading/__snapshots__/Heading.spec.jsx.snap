// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render as an H1 1`] = `
<Heading
  appearance="H1"
>
  <styled.h1>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "VCoKk",
            "rules": Array [
              "
  font-size: 2rem;
  line-height: 3rem;
  color: #162d3d;
",
            ],
          },
          "displayName": "styled.h1",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "h1",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h1
        className="sc-bdVaJa VCoKk"
      >
        Hello World
      </h1>
    </StyledComponent>
  </styled.h1>
</Heading>
`;

exports[`Heading should render as an H2 1`] = `
<Heading
  appearance="H2"
>
  <styled.h2>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "dTAZrU",
            "rules": Array [
              "
  font-size: 1.75rem;
  line-height: 2rem;
  color: #162d3d;
",
            ],
          },
          "displayName": "styled.h2",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "h2",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h2
        className="sc-bwzfXH dTAZrU"
      >
        Hello World
      </h2>
    </StyledComponent>
  </styled.h2>
</Heading>
`;

exports[`Heading should render as an H3 1`] = `
<Heading
  appearance="H3"
>
  <styled.h3>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "jiDSp",
            "rules": Array [
              "
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #162d3d;
",
            ],
          },
          "displayName": "styled.h3",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "h3",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h3
        className="sc-htpNat jiDSp"
      >
        Hello World
      </h3>
    </StyledComponent>
  </styled.h3>
</Heading>
`;

exports[`Heading should render as an H4 1`] = `
<Heading
  appearance="H4"
>
  <styled.h4>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "lastClassName": "fNeiKa",
            "rules": Array [
              "
  font-size: 1.125rem;
  line-height: 1.5rem;
  color: #162d3d;
",
            ],
          },
          "displayName": "styled.h4",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "h4",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h4
        className="sc-bxivhb fNeiKa"
      >
        Hello World
      </h4>
    </StyledComponent>
  </styled.h4>
</Heading>
`;

exports[`Heading should render as an H5 1`] = `
<Heading
  appearance="H5"
>
  <styled.h5>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "lastClassName": "kCOWXr",
            "rules": Array [
              "
  font-size: 0.75rem;
  line-height: 1.5rem;
  color: #32536a;
  text-transform: uppercase;
  letter-spacing: 1px;
",
            ],
          },
          "displayName": "styled.h5",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "h5",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h5
        className="sc-ifAKCX kCOWXr"
      >
        Hello World
      </h5>
    </StyledComponent>
  </styled.h5>
</Heading>
`;

exports[`Heading should render as an H6 1`] = `
<Heading
  appearance="H6"
>
  <styled.h6>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "lastClassName": "edLHym",
            "rules": Array [
              "
  font-size: 0.625rem;
  line-height: 1.125rem;
  color: #162d3d;
  text-transform: uppercase;
  letter-spacing: 1px;
",
            ],
          },
          "displayName": "styled.h6",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "h6",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h6
        className="sc-EHOje edLHym"
      >
        Hello World
      </h6>
    </StyledComponent>
  </styled.h6>
</Heading>
`;
